pex/environment.py

```diff
@@ -216,16 +216,18 @@ def mount(
         target=None,  # type: Optional[DistributionTarget]
     ):
         # type: (...) -> PEXEnvironment
-        pi = pex_info or PexInfo.from_pex(pex)
-        pex_hash = pi.pex_hash
+        if not pex_info:
+            pex_info = PexInfo.from_pex(pex)
+            pex_info.update(PexInfo.from_env())
```

Comment:
This is an unsatisfying fix, but there is now (previously failing) test coverage. A more wholistic re-factor of PexInfo / Variables loading is in order, but that's a delicate change that will need to be isolated and deliberated.